glusterd2/xlator/options/options.go

```diff
@@ -160,7 +160,23 @@ func ValidateBool(o *Option, val string) error {
 
 // ValidateClientAuthAddr validates mount auth address
 func ValidateClientAuthAddr(o *Option, val string) error {
-	return ErrInvalidArg
+	if len(val) == 0 {
+		return ErrInvalidArg
+	}
+	err := ValidateInternetAddress(o, val)
+	if err != nil {
+		i := strings.Index(val, "/")
+		if i != -1 && validate.IsIP(val[:i]) {
```

Comment:
You don't need custom validation functions for the subnet patterns, or the CIDR notation, as they're properly called. You should use `net.ParseCIDR`. Or better still use govalidators `IsCIDR`.

Reply:
@kshlm ok. So since ValidateInternetAddress() already uses net.ParseCIDR to check for subnet patterns, so just calling the same function to handle this validation as well.
Please review. thanks.